_aicspylibczi/ImageFactory.cpp

```diff
@@ -104,34 +128,37 @@ namespace pylibczi {
       shape.emplace_back(size.h);
       shape.emplace_back(size.w);
 
-      std::shared_ptr<Image> image = s_pixelToImage[pixelType](shape, pixelType, plane_coordinate_, box_, index_m_);
+      size_t mem_index = 0;
+
+      std::shared_ptr<Image> image = s_pixelToImage[pixelType](shape, pixelType, plane_coordinate_, box_,
+          m_imgContainer.get(), mem_index_, index_m_);
       if (image==nullptr)
           throw std::bad_alloc();
       image->loadImage(bitmap_ptr_, channels);
-
-      return std::shared_ptr<Image>(image);
+      auto sImage = std::shared_ptr<Image>(image);
+      m_imgContainer->addImage(sImage);
+      return sImage;
   }
 
-  Image::ImVec ImageFactory::splitToChannels(std::shared_ptr<Image> img_in_)
-  {
-      Image::ImVec ivec;
-      auto shape = img_in_->shape();
-      if (shape.size() != 3)
-          throw ImageSplitChannelException("TypedImage  only has 2 dimensions. No channels to split.", 0);
-      int cDim = 0;
-
-      // NOTE I'm forcing the user to specify C, basically because it's difficult to assign meaning between BGR
-      // values for multiple channels
-      for (int i = 0; i<shape[0]; i++) {
-          // TODO should I change the pixel type from a BGRx to a Grayx/3
-          libCZI::PixelType pt = img_in_->pixelType();
-          if(pt == libCZI::PixelType::Invalid){
-              img_in_->coordinatePtr()->TryGetPosition(libCZI::DimensionIndex::C, &cDim);
-              throw ImageSplitChannelException("Only PixelTypes Bgr24, Bgr48, and Bgr96Float can be split! "
-                                               "You have attempted to split an invalid PixelType", cDim);
+  std::vector< std::pair< char, size_t> >
+  ImageFactory::getFixedShape(void){
+      auto images = m_imgContainer->images();
+      images.sort();
+      auto charSizes = images.getShape();
+      if( m_imgContainer->pixelType() == libCZI::PixelType::Bgr24 ||
+          m_imgContainer->pixelType() == libCZI::PixelType::Bgr48 ||
+          m_imgContainer->pixelType() == libCZI::PixelType::Bgr96Float ){
+          auto ittr = find_if(charSizes.begin(), charSizes.end(), [](std::pair<char, size_t> &pr){
+              return (pr.first == 'C');
+          });
+          if( ittr != charSizes.end()) ittr->second *=3;
+          else {
+              charSizes.push_back(std::pair<char, size_t>('C', 3));
+              std::sort(charSizes.begin(), charSizes.end(), [&](std::pair<char, size_t> a_, std::pair<char, size_t> b_) {
+                  return libCZI::Utils::CharToDimension(a_.first)>libCZI::Utils::CharToDimension(b_.first);
+              });
```

Comment:
I would love more comments in general but specifically around here because next time we have bugs with BGR images this will be the first place to look I would assume.